python/examples/sample-selection.py

```diff
@@ -0,0 +1,74 @@
+import sys
+
+import chemfiles
+
+from rascaline import SoapPowerSpectrum
+from equistore import Labels, TensorMap
+import numpy as np
+
+with chemfiles.Trajectory(sys.argv[1]) as trajectory:
+    frames = [f for f in trajectory]
+
+HYPER_PARAMETERS = {
+    "cutoff": 5.0,
+    "max_radial": 6,
+    "max_angular": 4,
+    "atomic_gaussian_width": 0.3,
+    "center_atom_weight": 1.0,
+    "radial_basis": {
+        "Gto": {},
+    },
+    "cutoff_function": {
+        "ShiftedCosine": {"width": 0.5},
+    },
+}
+
+calculator = SoapPowerSpectrum(**HYPER_PARAMETERS)
+
+descriptor = calculator.compute(frames)
+
+# The names of the samples that will be chosen during training
+# coincide with the samples of the blocks obtained after calculating the
+# properties. Accordingly, the names of the samples can be obtained with the
+# following commands (usually 'structure' (structure number) and 'center'
+# (atom number in the structure).
+print(f"samples names: {descriptor.block(0).samples.names}")
+
+# selected_samples can be a variable of `Labels` or `TensorMap` type.
+# In the first case, if we want to select first four structures:
+labels = Labels(
+    names=["structure"],
+    values=np.array([[0], [1], [2], [3]]),
+)
+
+reference_descriptor = calculator.compute(frames, selected_samples=labels)
+
+# In the latter case, from the `TensorMap`, which we pass, the list
+# of samples is taken, which will be in the final result. The code
+# with looks as following:
+
+descriptor_for_comparison = calculator.compute(
+    frames, selected_samples=reference_descriptor
+)
+print(f"shape of first block initially: {descriptor.block(0).values.shape}")
+print(
+    f"shape of first block of reference: {reference_descriptor.block(0).values.shape}"
+)
+print(
+    f"shape of first block after selection: {descriptor_for_comparison.block(0).values.shape}"
+)
```

Comment:
This does not really demonstrate the advantage of using a TensorMap for selection (different selections for different keys). Could we do it differently?

Reply:
> This does not really demonstrate the advantage of using a TensorMap for selection (different selections for different keys). Could we do it differently?

Well, we can do feature selection block by block, then do FPS on the whole dataset and see which result is better (for the same number of features in the final matrix). Is that ok?

Reply:
this example is about sample selection. We could also only have a `TensorMap` selection for features selection, which will be easier to explain

Reply:
Ah, hmm. Well, we can try to output all the data block by block, but that would be more just an illustration of how the algorithm works. About feature_selection - it seems to me that there's nothing wrong with explaining a simple way of working. It doesn't clutter up the tutorial, but it gives some useful background.